static/js/components/OrderSummary_test.js

```diff
@@ -64,27 +64,32 @@ describe('OrderSummary', () => {
     );
   };
 
-  it('shows discount calculation if user has a coupon', () => {
-    let course = findCourse(course => (
-      course.runs.length > 0 &&
-      course.runs[0].status === STATUS_OFFERED
-    ));
-    let firstRun = course.runs[0];
-    const wrapper = renderOrderSummary({
-      courseRun: firstRun,
-      course: course,
-      discount: COURSE_PRICES_RESPONSE[1].price,
-      finalPrice: 0,
-      couponCode: '561KH'
+  for (let code of ['561KH', null, '']) {
+    it(`shows discount calculation if user has a coupon with code ${code}`, () => {
```

Comment:
I would recommend changing the loop and the description of the test cases to something like this:

```javascript
[
  ['561KH', 'non-blank coupon code'],
  [null, 'null coupon code'],
  ['', 'blank coupon code']
].forEach(([code, codeDescription]) => {
  it(`shows discount calculation if user has a coupon with ${codeDescription}`, () => {
    // ...    
  });
});
```